src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -202,6 +210,90 @@ private RocksDb Open(string path, (DbOptions Options, ColumnFamilies? Families)
+            {
+                DateTime creationTime = File.GetCreationTimeUtc(fullPath);
+                fileMetadatas.Add((liveFileMetadata.FileMetadata, creationTime));
+            }
+            catch (IOException)
+            {
+                // Maybe the file is gone or something. We ignore it.
+            }
+        }
+
+        fileMetadatas.Sort((item1, item2) =>
+        {
+            // Sort them by level so that lower level get priority
+            int levelDiff = item1.metadata.FileLevel - item2.metadata.FileLevel;
+            if (levelDiff != 0) return levelDiff;
+
+            // Otherwise, we pick which file is newest.
+            return item2.creationTime.CompareTo(item1.creationTime);
+        });
+
+        long totalSize = 0;
+        fileMetadatas = fileMetadatas.TakeWhile(metadata =>
+        {
+            availableMemory -= (long)metadata.metadata.FileSize;
+            bool take = availableMemory > 0;
+            if (take)
+            {
+                totalSize += (long)metadata.metadata.FileSize;
+            }
+            return take;
+        })
+            // We reverse them again so that lower level goes last so that it is the freshest.
+            // Not all of the available memory is actually available so we are probably over reading things.
+            .Reverse()
+            .ToList();
+
+        ConcurrentQueue<(FileMetadata metadata, DateTime creationTime)> taskQueues = new(fileMetadatas);
+
+        long totalRead = 0;
+        Task[] tasks = Enumerable.Range(0, Environment.ProcessorCount).Select((_) =>
+        {
+            return Task.Run(() =>
+            {
+                byte[] buffer = new byte[512.KiB()];
+                while (taskQueues.TryDequeue(out (FileMetadata metadata, DateTime creationTime) liveFileMetadata))
+                {
+                    string fullPath = Path.Join(basePath, liveFileMetadata.metadata.FileName);
+                    _logger.Info($"{(totalRead * 100 / (double)totalSize):00.00}% Warming up file {fullPath}");
+
+                    try
+                    {
+                        using FileStream stream = File.OpenRead(fullPath);
+                        int readCount = buffer.Length;
+                        while (readCount == buffer.Length)
+                        {
+                            readCount = stream.Read(buffer);
+                            Interlocked.Add(ref totalRead, readCount);
+                        }
+                    }
+                    catch (FileNotFoundException)
+                    {
+                        // Happens sometimes. We do nothing here.
+                    }
+                    catch (IOException e)
+                    {
+                        // Something unusual, but nothing noteworthy.
+                        _logger.Warn($"Exception warming up {fullPath} {e}");
+                    }
+                }
+            });
+        }).ToArray();
+        Task.WaitAll(tasks);
```

Comment:
Wouldn't Parallel.For be better?

Reply:
No idea. I've switched to `Parallel.ForEach`.